Tests/SwiftDiagnosticsTest/DiagnosticFormatterTests.swift

```diff
@@ -0,0 +1,31 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftDiagnostics
+import XCTest
+
+final class DiagnosticFormatterTests: XCTestCase {
+  func testFormattedMessage() {
+    let message = SimpleDiagnosticMessage(
+      message: "something went wrong",
+      diagnosticID: MessageID(domain: "swift-syntax", id: "testing"),
+      severity: .error,
+      category: DiagnosticCategory(
+        name: "Testing",
+        documentationURL: "http://example.com"
+      )
+    )
+
+    let formattedText = DiagnosticsFormatter().formattedMessage(message)
+    XCTAssertEqual(formattedText, "error: something went wrong [#Testing]")
```

Comment:
Is the idea that the caller of this function will be responsible for emitting the link between `[#Testing]` and `http://example.com`?

Reply:
If your terminal supports it, that `[#Testing]` will be a hyperlink. There's also existing API for emitting the footnote style for a set of categoryies on DiagnosticsFormatter:

```swift
public func categoryFootnotes(
    _ categories: some Sequence<DiagnosticCategory>,
    leadingText: String = "\n"
  ) -> String
```

because you don't want to emit the footnotes after every diagnostic, you want to do it once at the end.

Reply:
Oh, great. I didn’t realize that we had an API for the footnotes. 👍🏽